bootstrap/bukkit/src/main/java/org/geysermc/platform/bukkit/world/GeyserBukkitWorldManager.java

```diff
@@ -28,18 +28,43 @@
 
 import com.github.steveice10.mc.protocol.data.game.world.block.BlockState;
 
+import lombok.AllArgsConstructor;
 import org.bukkit.Bukkit;
+import org.bukkit.block.Block;
 import org.geysermc.connector.network.session.GeyserSession;
 import org.geysermc.connector.network.translators.world.WorldManager;
 import org.geysermc.connector.network.translators.world.block.BlockTranslator;
+import us.myles.ViaVersion.protocols.protocol1_13_1to1_13.Protocol1_13_1To1_13;
+import us.myles.ViaVersion.protocols.protocol1_15to1_14_4.data.MappingData;
 
+@AllArgsConstructor
 public class GeyserBukkitWorldManager extends WorldManager {
 
+    // Since ProtocolSupport isn't supported (at least with Floodgate) as of the time of this code, you need ViaVersion to connect to an older server.
+    // However, there is a possibility that Geyser can connect to a server without ViaVersion on Spigot, so this is the precaution
+    private final boolean isViaVersion;
+
     @Override
+    @SuppressWarnings("deprecation")
     public BlockState getBlockAt(GeyserSession session, int x, int y, int z) {
         if (session.getPlayerEntity() == null) {
             return BlockTranslator.AIR;
         }
-        return BlockTranslator.getJavaIdBlockMap().get(Bukkit.getPlayer(session.getPlayerEntity().getUsername()).getWorld().getBlockAt(x, y, z).getBlockData().getAsString());
+        try {
+            return BlockTranslator.getJavaIdBlockMap().get(Bukkit.getPlayer(session.getPlayerEntity().getUsername()).getWorld().getBlockAt(x, y, z).getBlockData().getAsString());
+        } catch (NoSuchMethodError ex) {
```

Comment:
Might be better to check the version directly rather than a try/catch each time this method is called.

Reply:
There's no good way to check the version, since all "get version" methods output strings and I certainly wouldn't want to test for every possible combination of "1.8/1.9/1.10...". I don't like the `try/catch` here either and would rather replace it with something else.